src/network/network_gui.cpp

```diff
@@ -138,10 +138,12 @@ class NWidgetServerListHeader : public NWidgetContainer {
 		this->current_x = given_width;
 		this->current_y = given_height;
 
-		given_width -= this->tail->smallest_x;
+		given_width -= this->children.back()->smallest_x;
 		/* The first and last widget are always visible, determine which other should be visible */
-		for (NWidgetBase *child_wid = this->head->next; child_wid->next != nullptr; child_wid = child_wid->next) {
-			if (given_width > ScaleGUITrad(MINIMUM_NAME_WIDTH_BEFORE_NEW_HEADER) + child_wid->smallest_x && child_wid->prev->current_x != 0) {
+		auto end = std::prev(std::end(this->children));
+		for (auto it = std::next(std::begin(this->children)); it != end; ++it) {
```

Comment:
apart from being very ugly (though i'm not sure how that could be improved), this also has the potential issue of a one-length array never terminating (it = container.end, end = container.0th) . Is that worth worrying about?

Reply:
It's not like the original was safe for a one-length array: `child_wid` would be `nullptr` because `this->head->next` must be `nullptr` for a one-length array. Then the condition of the for loop would do a nullptr-dereference.

I'm seeing asserts below to check for a specific `size()` of children, might it be useful to add something like `assert(this->children.size() >= 2);`?

Reply:
Due to the nature of this widget, it's not possible, but I have added the check for size() > 2 anyway.